server/storage/storage_impl.go

```diff
@@ -50,87 +54,177 @@ func newEtcdStorage(client *clientv3.Client, rootPath string, opts Options) Stor
 }
 
 func (s *metaStorageImpl) CreateCluster(ctx context.Context, cluster *clusterpb.Cluster) (*clusterpb.Cluster, error) {
-	return nil, nil
+	now := time.Now()
+	cluster.CreatedAt = uint64(now.Unix())
+
+	value, err := proto.Marshal(cluster)
+	if err != nil {
+		return nil, ErrParseCreateCluster.WithCausef("fail to parse proto，clusterID:%d, err:%v", cluster.Id, err)
+	}
+
+	key := path.Join(s.rootPath, makeClusterKey(cluster.Id))
+
+	// Check if the key exists, if not，create cluster; Otherwise, the cluster already exists and return an error.
+	cmp := clientv3util.KeyMissing(key)
+	opCreateCluster := clientv3.OpPut(key, string(value))
+
+	resp, err := s.Txn(ctx).
+		If(cmp).
+		Then(opCreateCluster).
+		Commit()
+	if err != nil {
+		return nil, errors.Wrapf(err, "fail to create cluster, clusterID:%d, key:%s", cluster.Id, key)
+	}
+	if !resp.Succeeded {
+		return nil, ErrParseCreateCluster.WithCausef("cluster may already exist, clusterID:%d, key:%s, resp:%v", cluster.Id, key, resp)
+	}
+	return cluster, nil
 }
 
 func (s *metaStorageImpl) GetCluster(ctx context.Context, clusterID uint32) (*clusterpb.Cluster, error) {
 	key := makeClusterKey(clusterID)
 	value, err := s.Get(ctx, key)
 	if err != nil {
-		return nil, errors.Wrapf(err, "meta storage get cluster failed, clusterID:%d, key:%s", clusterID, key)
+		return nil, errors.Wrapf(err, "fail to get cluster, clusterID:%d, key:%s", clusterID, key)
 	}
 
-	meta := &clusterpb.Cluster{}
-	if err = proto.Unmarshal([]byte(value), meta); err != nil {
-		return nil, ErrParseGetCluster.WithCausef("proto parse failed, clusterID:%d, err:%s", clusterID, err)
+	cluster := &clusterpb.Cluster{}
+	if err = proto.Unmarshal([]byte(value), cluster); err != nil {
+		return nil, ErrParseGetCluster.WithCausef("fail to parse proto, clusterID:%d, err:%v", clusterID, err)
 	}
-	return meta, nil
+	return cluster, nil
 }
 
-func (s *metaStorageImpl) CreateClusterTopology(ctx context.Context, clusterTopology *clusterpb.ClusterTopology) error {
-	return nil
-}
-
-func (s *metaStorageImpl) PutCluster(ctx context.Context, clusterID uint32, meta *clusterpb.Cluster) error {
-	value, err := proto.Marshal(meta)
+func (s *metaStorageImpl) PutCluster(ctx context.Context, clusterID uint32, cluster *clusterpb.Cluster) error {
+	value, err := proto.Marshal(cluster)
 	if err != nil {
-		return ErrParsePutCluster.WithCausef("proto parse failed, clusterID:%d, err:%s", clusterID, err)
+		return ErrParsePutCluster.WithCausef("fail to parse proto, clusterID:%d, err:%v", clusterID, err)
 	}
 
 	key := makeClusterKey(clusterID)
 	err = s.Put(ctx, key, string(value))
 	if err != nil {
-		return errors.Wrapf(err, "meta storage put cluster failed, clusterID:%d, key:%s", clusterID, key)
+		return errors.Wrapf(err, "fail to put cluster, clusterID:%d, key:%s", clusterID, key)
 	}
 	return nil
 }
 
+func (s *metaStorageImpl) CreateClusterTopology(ctx context.Context, clusterTopology *clusterpb.ClusterTopology) (*clusterpb.ClusterTopology, error) {
```

Comment:
Add some comments to explain this method behavior especially:
```
Return error if the topology already exists.
```

BTW, the returned topology seems unnecessary.

Reply:
The returned topology has the creation time